measurement/plugins/ip_route/measurements.py

```diff
@@ -0,0 +1,116 @@
+"""
+ip_route measurement, formerly the Traceroute measurement.
+In a similar manner to the download_test, a list of hosts can be passed, in which case the one with the lowest latency is chosen.
+The results are returned in order as:
+ - IPRouteMeasurementResult
+ - LatencyResult of the host used to create the IPRouteMeasurement
+ - The LatencyResults from determining the initial latency
+Traceroute measurement is provided by the "scapy" library. Note that this library needs route privileges to create
+raw sockets, or to have the CAP_NET_RAW capability set.
+
+"""
+import socket
+import validators
+from validators import ValidationFailure
+
+import scapy
+
+from measurement.measurements import BaseMeasurement
+from measurement.results import Error
+from measurement.plugins.ip_route.results import IPRouteMeasurementResult
+from measurement.plugins.latency.measurements import LatencyMeasurement
+
+ROUTE_ERRORS = {
+    "route-err": "iproute encountered an unknown error",
+    "route-address": "iproute failed to find socket address",
+    "route-permission": "iproute does not have permission to create a raw socket",
+}
+
+
+class IPRouteMeasurement(BaseMeasurement):
+    def __init__(self, id, hosts, route_timeout=10, count=4):
+        super(IPRouteMeasurement, self).__init__(id=id)
+
+        if len(hosts) < 1:
+            raise ValueError("At least one url must be provided.")
+        for host in hosts:
+            validated_url = validators.domain(host)
+            if isinstance(validated_url, ValidationFailure):
+                raise ValueError("`{host}` is not a valid host".format(host=host))
```

Comment:
It's common to traceroute to an IP address like `1.1.1.1`. Currently that doesn't work:

```
In [4]: from measurement.plugins.ip_route.measurements import IPRouteMeasurement                                                                                                                                   

In [5]: irm = IPRouteMeasurement( 
   ...:     "{uuid}".format(uuid=uuid.uuid4()), 
   ...:     ["1.1.1.1", "8.8.8.8"] 
   ...:     )                                                                                                                                                                                                      
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-48710f25826c> in <module>
      1 irm = IPRouteMeasurement(
      2     "{uuid}".format(uuid=uuid.uuid4()),
----> 3     ["1.1.1.1", "8.8.8.8"]
      4     )

~/dev/honestybox/honestybox-measurement/measurement/plugins/ip_route/measurements.py in __init__(self, id, hosts, route_timeout, count)
     37             validated_url = validators.domain(host)
     38             if isinstance(validated_url, ValidationFailure):
---> 39                 raise ValueError("`{host}` is not a valid host".format(host=host))
     40 
     41         if count < 0:

ValueError: `1.1.1.1` is not a valid host
```